File: backend/src/main/java/com/wellbeignatwork/backend/entity/Evaluation/VoteIdea.java
package com.wellbeignatwork.backend.entity.Evaluation;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class VoteIdea implements Serializable {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idVote;
    private  int nbYes;
    private  int nbNo;

    @ManyToOne
    @JoinColumn(name = "idUser", referencedColumnName = "USER_ID")
    @JsonIgnore
    User idUser;

    @ManyToOne
    @JoinColumn(name = "idSujet", referencedColumnName = "id")

    @JsonIgnore
    Sujet idSujet;




}

File: backoffice/src/app/Models/Evaluation/VoteIdea.ts
import {Sujet} from "./Sujet";
import { User } from "../../evaluation/user";

export class VoteIdea {
  idVote:number;
  nbYes:number;
  nbNo: number;
  idUser:User;
  idSujet:Sujet;
}
export interface IVote {
  nbNo:number,
  nbYes:number
}
